src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrayCopy = [...this];
```

Comment:
you don't need to create a copy of the array

Reply:
If I don't create copy I'll change the origin input